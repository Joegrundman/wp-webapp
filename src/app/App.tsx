// import Header from '../components/header/Header'
import Header from 'Components/header/Header'
import * as React from 'react'
import GameBoard from 'Views/game/Game'
import locals from './App.css'

class App extends React.Component {
  public render(): JSX.Element {
    return (
      <div className={locals.App}>
        <Header />
        <GameBoard />
      </div>
    );
  }
}

export default App
