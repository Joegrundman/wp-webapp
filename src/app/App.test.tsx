import { shallow, ShallowWrapper } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import * as React from 'react';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

it('renders an app component', () => {
  const component: ShallowWrapper = shallow(<App/>);
  expect(component).toMatchSnapshot();
});
