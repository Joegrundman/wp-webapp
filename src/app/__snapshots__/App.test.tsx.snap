// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an app component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className={undefined}
        >
          <img
            alt="logo"
            className={undefined}
            src={undefined}
          />
          <h1
            className={undefined}
          >
            Welcome to React
          </h1>
        </header>,
        <p
          className={undefined}
        >
          To get started, edit 
          <code>
            src/App.tsx
          </code>
           and save to reload.
        </p>,
        <GameBoard />,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className={undefined}
              src={undefined}
            />,
            <h1
              className={undefined}
            >
              Welcome to React
            </h1>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome to React",
              "className": undefined,
            },
            "ref": null,
            "rendered": "Welcome to React",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "To get started, edit ",
            <code>
              src/App.tsx
            </code>,
            " and save to reload.",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          "To get started, edit ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "src/App.tsx",
            },
            "ref": null,
            "rendered": "src/App.tsx",
            "type": "code",
          },
          " and save to reload.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className={undefined}
          >
            <img
              alt="logo"
              className={undefined}
              src={undefined}
            />
            <h1
              className={undefined}
            >
              Welcome to React
            </h1>
          </header>,
          <p
            className={undefined}
          >
            To get started, edit 
            <code>
              src/App.tsx
            </code>
             and save to reload.
          </p>,
          <GameBoard />,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className={undefined}
                src={undefined}
              />,
              <h1
                className={undefined}
              >
                Welcome to React
              </h1>,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": undefined,
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to React",
                "className": undefined,
              },
              "ref": null,
              "rendered": "Welcome to React",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "To get started, edit ",
              <code>
                src/App.tsx
              </code>,
              " and save to reload.",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            "To get started, edit ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "src/App.tsx",
              },
              "ref": null,
              "rendered": "src/App.tsx",
              "type": "code",
            },
            " and save to reload.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
